services/alerts/src/makeAlertsManager.ts

```diff
@@ -0,0 +1,38 @@
+import { Alert, AlertsManager, AlertsManagerAlert } from './types'
+
+type SetAlertsFunction = React.Dispatch<
+    React.SetStateAction<AlertsManagerAlert[]>
+>
+
+const createAlertManagerAlert = (
+    alert: Alert,
+    id: number,
+    remove: Function
+): AlertsManagerAlert => ({
+    ...alert,
+    id,
+    remove: () => remove(id),
+})
+
+export const makeAlertsManager = (
+    setAlerts: SetAlertsFunction
+): AlertsManager => {
+    let id = 0
+
+    const remove = (id: number) => {
+        setAlerts((alerts: AlertsManagerAlert[]) =>
+            alerts.filter(alert => alert.id !== id)
+        )
+    }
+    const add = (alert: Alert) => {
+        setAlerts(alerts => [
+            ...alerts,
+            createAlertManagerAlert(alert, ++id, remove),
```

Comment:
I hadn't really considered the starting number here, but changed `id++` to `++id` so that it would match the previous implementation which first called `id++` and used it later.

But to me it also makes sense to start at `1` for the reasons you outline, and most databases do the same... So keeping this as is and resolving the conversation.